### PR TITLE
`fn add_compound_extended_candidate`: Cleanup and make safe

### PR DESCRIPTION
**Pull request by kkysen**

### CODE REVIEW

File: src/refmvs.rs [hunk 1]
```diff
@@ -486,115 +486,76 @@ unsafe extern "C" fn add_temporal_candidate(
...
-        if cand_ref <= 0 as libc::c_int {
+    let (same, diff) = same.split_at_mut(2);
+
+    // Alternative way with compile-time index checking
+    // for the resulting array refs (rather than slices).
+    //
+    // Safety: `[usize; 4]` and `[[usize; 2]; 2]` should be identical.
+    // let [same_count, diff_count] = unsafe { 
+    //     std::mem::transmute::<&mut [usize; 4], &mut [[usize; 2]; 2]>(same_count)
+    // };
```
Comment (randomPoison) [inline src/refmvs.rs:507]: Do we need this comment? Seems like it's documenting a different approach that you're no longer using?

**Review by randomPoison — APPROVED** [review]

File: src/refmvs.rs (hunk 1 above)
Comment (kkysen) [reply src/refmvs.rs:507]: Yeah, I added it in case we want to implement it that way (trading a transmute for compile-time indexing).  Should I remove it?

Comment (randomPoison) [reply src/refmvs.rs:507]: I think that's an approach that would be cool to take once safe transmute becomes a thing, but probably not worth the extra unsafe operation when `split_at` already does the trick nicely. So I'd say remove the comment here.

Comment (kkysen) [reply src/refmvs.rs]: Removed in 5ea222199003cae8abd46c2b7236c3e11859c067.

**Pull request merged**